components/sensor/max17043.rst

```diff
@@ -0,0 +1,104 @@
+Analog Devices MAX17043 battery fuel gauge
+==========================================
+
+.. seo::
+    :description: Instructions for setting up Analog Devices MAX17043 battery fuel gauge in ESPHome.
+    :image: max17043.jpg
+
+The MAX17043 component allows you to use a MAX17043 to more accurately monitor the remaining capacity
+of a LIPO battery
+(`datasheet <https://www.analog.com/en/products/max17043.html#documentation>`__,
+`DFRobot`_) in ESPHome. It uses :ref:`I²C Bus <i2c>` for communication (the address is fixed at 0x36).
+
+Once configured, it uses a sophisticated Li+ battery-modeling scheme, called ModelGauge™ to track the
+battery's relative state-of-charge continuously over a widely varying charge/discharge profile.
+Unlike traditional fuel gauges, the ModelGauge algorithm eliminates the need for battery relearn cycles
+and an external current-sense resistor.
+
+In low power applications, it's very important to report battery levels accurately. By utilising ESPHome's ``deep_sleep``
+component together with a MAX17043, projects can run for extended periods and the user can be confident of the amount of battery remaining.
+This overcomes the reality that measured battery voltage does not correlate well at all with remaining battery charge.
+
+.. note::
+
+    See hardware design discussion below - it's important to leave the MAX17043 powered on during deep sleep.
+
+.. figure:: images/max17043.jpg
+    :align: center
+    :width: 60.0%
+
+    DFRobot DFR0563 Gravity I2C 3.7V Li battery fuel gauge.
+
+.. _DFRobot: https://www.dfrobot.com/product-1734.html
+
+Configuration
+*************
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: max17043
+        id: max17043_id
+        i2c_id: i2c_max17043
+        battery_voltage:
+          name: "Battery Voltage"
+        battery_level:
+          name: "Battery"
+
+Sensors
+*******
+
+- **battery_voltage** (*Optional*, float): The voltage measured at the LIPO battery.
+- **battery_level** (*Optional*, float): The percentage of battery remaining using the ModelGauge™ battery-modelling scheme.
```

Comment:
I think these should each have an "All other options from sensor..." as we do for other, similar sensors. Take a look at the [BSEC2 for BME680](https://esphome.io/components/sensor/bme68x_bsec2#sensor).